main.nf

```diff
@@ -83,12 +85,17 @@ process Build_config{
 
     publishDir params.outdir, mode: "copy"
 
+    input:
+        val(dir)
+        file(zarr_dirs)
+
     output:
         file("config.json")
 
     script:
+    concat_zarr_dirs = zarr_dirs.join(',')
     """
-    build_config.py --outdir ${params.outdir}
+    build_config.py --dataset ${params.dataset} --files_dir ${dir} --zarr_dirs ${concat_zarr_dirs}
```

Comment:
just a syntax suggestion: you can directly use the `params.dataset` in the process as you do here. But a cleaner way of doing this is to declare this variable as an input. And then use the internal variable name here. Just to make the I/O of each process clearer .

Reply:
followed your suggestion and declared it as an input